tap_redshift/resolve.py

```diff
@@ -89,11 +89,12 @@ def resolve_catalog(discovered, catalog, state):
 
         # These are the columns we need to select
         columns = desired_columns(selected, discovered_table.schema)
+        ordered_columns = list(filter(lambda x: x in columns, catalog_entry.schema.properties.keys()))
 
```

Comment:
The returned value of columns from desired_columns function was type of set, distorting the order of columns.